### ISSUE TITLE
Certbot failling

### ISSUE THREAD

**Issue by Flo451** [grindsa/acme2certifier#61]
Hi,

I have been having issues with Certbot requesting a certificate when the ACME server (0.15) is Django/MySQL backed. 
Interestingly acme.sh is working fine. 

Certbot logs

`sudo certbot certonly --server https://xxx --standalone --preferred-challenges http -d certbot-1.example.com -d certbot-2.example.com --cert-name certbot-test
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator standalone, Installer None
Requesting a certificate for certbot-1.example.com and certbot-2.example.com
An unexpected error occurred:
DeserializationError: Deserialization error: Could not decode 'challenges' ([{u'url': u'https://xxx/acme/chall/iiMvleTMIAeK', u'token': u'AkhhTaIxIRjw1b7a8mfH5xSERRihhpID', u'type': u'http-01'}, {u'url': u'https://xxx/acme/chall/VRpdW3qjyLxc', u'token': u'AkhhTaIxIRjw1b7a8mfH5xSERRihhpID', u'type': u'dns-01'}, {}]): Deserialization error: missing type field
Please see the logfiles in /var/log/letsencrypt for more details.
`

Server logs

`[Wed May 12 12:17:40.484018 2021] [wsgi:error] [pid 16:tid 140089995077376] [remote 10.131.22.80:58452] 10.131.22.80 /acme/newnonce {'header': {'Replay-Nonce': '- modified -'}}
[Wed May 12 12:17:40.587852 2021] [wsgi:error] [pid 16:tid 140089963607808] [remote 10.131.22.80:58452] 10.131.22.80 /acme/neworders {'header': {'Replay-Nonce': '- modified -'}}
[Wed May 12 12:17:40.687221 2021] [wsgi:error] [pid 16:tid 140089995077376] [remote 10.131.22.80:58452] acme2certifier database error in Challenge._new(): (1406, "Data too long for column 'type' at row 1")
[Wed May 12 12:17:40.692447 2021] [wsgi:error] [pid 16:tid 140089995077376] [remote 10.131.22.80:58452] 10.131.22.80 /acme/authz/ILRKCOcPKNVD {'data': {'expires': '2021-05-13T12:17:40Z', 'status': 'pending', 'identifier': {'type': 'dns', 'value': 'certbot-1.example.com'}, 'challenges': [{'type': 'http-01', 'url': 'https://xxx/acme/chall/iiMvleTMIAeK', 'token': '- modified - '}, {'type': 'dns-01', 'url': 'https://xxx/acme/chall/VRpdW3qjyLxc', 'token': '- modified - '}, {}]}, 'code': 200, 'header': {'Replay-Nonce': '- modified -'}}`

**Comment by grindsa** [comment]
`Data too long for column 'type' at row 1` looks like something stupid on my side.... 

Is this problem reproducible? If so can you enable debug and share the debug log? Any chance to share the `/var/log/letsencrypt` file?

**Comment by Flo451** [comment]
Thanks, I was just about to provide the logs but seems you were faster!

**Comment by grindsa** [comment]
Hi,

Fix got included in [0.15.3](https://github.com/grindsa/acme2certifier/releases/tag/0.15.3).  If you are not using the container version it should be enough to update your `acme/models.py` with the [version from the fix package](https://raw.githubusercontent.com/grindsa/acme2certifier/master/examples/django/acme/models.py) and run `makemigrations` and `migrate` commands.

The issue got introduced with 0.15.  The challenge type "tls-alpn-01" could not get written to the `challenge` table as the field length was too short.  Thus, an empty json object was part of the challenge response which led to the observed  deserialization error in certbot.

```
Deserialization error: Could not decode 'challenges' ([..., {u'url': u'https://xxx/acme/chall/VRpdW3qjyLxc', u'token': u'AkhhTaIxIRjw1b7a8mfH5xSERRihhpID', u'type': u'dns-01'}, {}]):
```

Seems acme.sh was able to handle this for some reason.  We will extend our regression suite to cover a mysql backend text in the next few days.

**Comment by Flo451** [comment]
Thank you, I verified the fix, seems good now. Certbot is able to request certs again. 

To be honest I was not able to get the Django migration going with makemigrations, it would always report ("no changes detected") even after updating models.py. I even specified the app name as "acme" for the migration but no luck.

In the end I just altered the table with SQL extending the varchar column to 15.

Thanks!
